src/sensors/bno080sensor.cpp

```diff
@@ -117,39 +129,27 @@ void BNO080Sensor::motionLoop()
 #if USE_6_AXIS
         if (imu.hasNewGameQuat()) // New quaternion if context
         {
-            imu.getGameQuat(quaternion.x, quaternion.y, quaternion.z, quaternion.w, calibrationAccuracy);
-            quaternion *= sensorOffset;
+            imu.getGameQuat(fusedRotation.x, fusedRotation.y, fusedRotation.z, fusedRotation.w, calibrationAccuracy);
+            fusedRotation *= sensorOffset;
 
-    #if ENABLE_INSPECTION
-            {
-                Network::sendInspectionFusedIMUData(sensorId, quaternion);
-            }
-    #endif // ENABLE_INSPECTION
-
-            if (!OPTIMIZE_UPDATES || !lastQuatSent.equalsWithEpsilon(quaternion))
+            if (ENABLE_INSPECTION || !OPTIMIZE_UPDATES || !lastFusedRotationSent.equalsWithEpsilon(fusedRotation))
             {
-                newData = true;
-                lastQuatSent = quaternion;
+                newFusedRotation = true;
+                lastFusedRotationSent = fusedRotation;
             }
             // Leave new quaternion if context open, it's closed later
 
 #else // USE_6_AXIS
 
         if (imu.hasNewQuat()) // New quaternion if context
         {
-            imu.getQuat(quaternion.x, quaternion.y, quaternion.z, quaternion.w, magneticAccuracyEstimate, calibrationAccuracy);
-            quaternion *= sensorOffset;
+            imu.getQuat(fusedRotation.x, fusedRotation.y, fusedRotation.z, fusedRotation.w, magneticAccuracyEstimate, calibrationAccuracy);
+            fusedRotation *= sensorOffset;
 
-    #if ENABLE_INSPECTION
+            if (ENABLE_INSPECTION || !OPTIMIZE_UPDATES || !lastFusedRotationSent.equalsWithEpsilon(fusedRotation))
```

Comment:
```suggestion
            if (!OPTIMIZE_UPDATES || !lastFusedRotationSent.equalsWithEpsilon(fusedRotation))
```

Reply:
> There's still inspection code left over, please remove that as well

This PR does not remove all inspection code, it removes redundand inspection code like sending IMU data separately when it's already sent.

Reply:
Look at the if-statement again, it will break the code if you leave this in here

Reply:
this should make it always send updates when inspection is enabled, even if optimization is enabled and quaternion didn't change substantially

Reply:
Nevermind, you're right